src/marqo/marqo_cloud_instance_mappings.py

```diff
@@ -25,21 +26,27 @@ def get_control_base_url(self) -> str:
         return f"{self._control_base_url}/api"
 
     def get_index_base_url(self, index_name: str) -> str:
+        """Returns the index_name's base URL regardless of its status.
+
+        Raises:
+            MarqoCloudIndexNotFoundError: if index_name is not found in any status.
+        """
         self._refresh_urls_if_needed(index_name)
-        if index_name in self._urls_mapping[IndexStatus.READY]:
-            return self._urls_mapping[IndexStatus.READY][index_name]
-        if index_name in self._urls_mapping[IndexStatus.CREATING]:
-            raise MarqoCloudIndexNotReadyError(index_name)
+
+        for cloud_status, indexes in self._urls_mapping.items():
+            if index_name in indexes:
+                return indexes[index_name]
+
         raise MarqoCloudIndexNotFoundError(index_name)
 
     def is_remote(self):
         return True
 
-    def _refresh_urls_if_needed(self, index_name):
-        if index_name not in self._urls_mapping[IndexStatus.READY] and \
-                time.time() - self.latest_index_mappings_refresh_timestamp > self.url_cache_duration:
-            # fast refresh to catch if index was created
-            self._refresh_urls()
+    def _refresh_urls_if_needed(self, index_name: Optional[str] = None):
+        if index_name is None or index_name not in self._urls_mapping[IndexStatus.READY]:
+            if time.time() - self.latest_index_mappings_refresh_timestamp > self.url_cache_duration:
+                # fast refresh to catch if index was created
```

Comment:
Comment doesn't apply anymore. Not a fast refresh

Reply:
actioned